app/src/functions/notify-endpoint-usage/app.ts

```diff
@@ -1,19 +1,35 @@
-import Redis from 'ioredis'
-import { influx, buildAppUsageQuery } from '../../lib/influx'
-import { getUTCTimestamp, getHoursFromNowUtcDate } from '../../lib/date-utils'
+import { influx, buildAppUsageQuery } from "../../lib/influx";
+import { getUTCTimestamp, getHoursFromNowUtcDate } from "../../lib/date-utils";
+import { GetUsageDataQuery } from "../../models/types";
+import { getApplicationNetworkData } from "../../lib/pocket";
 
-const redisHost = process.env.REDIS_HOST || ''
-const redisPort = process.env.REDIS_PORT || ''
+const redisHost = process.env.REDIS_HOST || "";
+const redisPort = process.env.REDIS_PORT || "";
 
-export async function getUsageData() {
-  const usage = influx.collectRows(buildAppUsageQuery({
-    start: getHoursFromNowUtcDate(0),
-    stop: getUTCTimestamp()
-  }))
+export async function getUsageData(): Promise<GetUsageDataQuery[]> {
+  const usage = (await influx.collectRows(
+    buildAppUsageQuery({
+      start: getHoursFromNowUtcDate(1),
+      stop: getUTCTimestamp(),
+    })
+  )) as unknown as any[];
 
-  return usage
+  const appData = usage.map((data: any) => ({
+    relays: data._value,
+    applicationPublicKey: data.applicationPublicKey,
+    result: data.result,
+    table: data.table,
+  }));
+
+  return appData as GetUsageDataQuery[];
 }
 
 exports.handler = async () => {
-  const redis = new Redis(parseInt(redisPort), redisHost)
+  const usage = await getUsageData();
+
+  const apps = Promise.allSettled(
+    usage.map((app) => getApplicationNetworkData(app.applicationPublicKey))
+  );
```

Comment:
This is probably fine and I know we'll build it up further, but pocket nodes (at least the ones we're using) like to error out _a lot_ when getting data from the network. While we first need to run tests, it's probable that we need to implement a simple retry mechanism (exponential backoff would probably work fine here). Either that or just error out and not care 🤷